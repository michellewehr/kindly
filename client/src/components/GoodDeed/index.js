import { useState } from "react";
import CommentForm from "../CommentForm";
import CommentsList from "../CommentsList";
import Auth from '../../utils/auth';

export default function GoodDeed({ goodDeed }) {
  const [viewComments, setViewComments] = useState(false);
  const [addComment, setAddComment] = useState(false);
  // only in profile
  // if (!goodDeeds.length) {
  //   return (
  //     <div>
  //       <h3>No Good Deeds yet!</h3>
  //     </div>
  //   );
  // }

  return (
    <div className="goodDeed w-full">
      <div className="rounded-lg mt-2 shadow-lg bg-white	 w-full flex flex-row flex-wrap p-3 antialiased">
        <div className="md:w-2/3 w-full px-3 flex flex-row flex-wrap">
          <div className="w-full text-gray-700 font-semibold relative pt-3 md:pt-0">
            <div className="flex flex-row text-2xl text-amber-500 leading-tight pb-1">
              {goodDeed.title}
            </div>
            <div className="text-sm text-amber-500 md:absolute pt-3 md:pt-0 top-0 right-0">
              {/* //!we need to be able to add this to the users total on */}
              verification Kindly Points: <b>+5</b>
            </div>
            <div className="text-normal hover:text-cyan-700 cursor-pointer text-cyan-900 pb-4">
              {/* //! get good deed host */}
              <span className="pb-1">{goodDeed.host.firstName} {goodDeed.host.lastName}</span>
            </div>
            <div className="text-normal text-cyan-900 pb-1">
              <span className="">
                {/* //! get good deed description */}
                <b>Description:</b> {goodDeed.deedText}
              </span>
            </div>
            <div className="text-normal text-cyan-900 pb-1">
              <span className="">
                {/* //! get good deed location */}
                <b>Location:</b>{goodDeed.location}
              </span>
            </div>
            {/* <div className="text-normal text-cyan-900 pb-1">
              <span className="">
                <b>Created At:</b> {goodDeed.createdAt}
              </span>
            </div> */}

            <div>
              {Auth.loggedIn() && !viewComments ? <button onClick={() => { setViewComments(true) }}>View Comments</button> : <button onClick={() => { setViewComments(false) }}>Hide Comments</button>}
            </div>
            <div>
              {Auth.loggedIn() &&
                <button onClick={() => { setAddComment(true) }}>Add Comment</button>}
            </div>
          </div>
          <div className="text-sm text-amber-500 md:absolute pt-3 md:pt-0 bottom-0 right-0">
            <button className="bg-cyan-700  hover:bg-orange-300 text-white font-bold py-2 px-4 rounded mt-1">
              Be Kind
              {/* //! need to add helper after signup */}
            </button>
          </div>
        </div>
      </div>
      {addComment && <CommentForm key={goodDeed._id} goodDeedId={goodDeed._id} />}
      {viewComments && <CommentsList comments={goodDeed.comments} key={goodDeed._id} />}

    </div>
  );
}
