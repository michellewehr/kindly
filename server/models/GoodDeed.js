const { Schema, model } = require('mongoose');
const dateFormat = require('../utils/dateFormat');
const commentSchema = require('./Comment');

const goodDeedSchema = new Schema(
  {
    host: {
      type: Schema.Types.ObjectId,
      ref: 'User'
    },
    title: {
      type: String,
      required: true,
      trim: true
    },
    helpers: {
      type: Schema.Types.ObjectId,
      ref: 'User'
    },
    date: {
      type: Date,
      required: true
    },
    deedText: {
      type: String,
      required: true,
      trim: true
    },
    location: {
      type: String,
      trim: true
    },
    createdAt: {
      type: Date,
      default: Date.now,
      get: timestamp => dateFormat(timestamp)
    },
    comments: [
      {
        type: Schema.Types.ObjectId,
        ref: 'Comment'
      }
    ],
    likes: {
      type: Number,
      default: 0
    }
  },
  {
    toJSON: {
      getters: true,
      virtuals: true
    }
  }
)

const GoodDeed = model('GoodDeed', goodDeedSchema);

//TODO: view count/ comment count?

module.exports = GoodDeed;
