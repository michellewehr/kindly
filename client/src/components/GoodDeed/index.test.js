import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import GoodDeed from './index';

afterEach(cleanup);

describe('GoodDeed component renders', () => {

   it('renders', () => {
      render(<GoodDeed />);
   });

   it('Matches snapshot DOM node structure', () => {
      const { asFragment } = render(<GoodDeed />);
      expect(asFragment()).toMatchSnapshot();
   });
});