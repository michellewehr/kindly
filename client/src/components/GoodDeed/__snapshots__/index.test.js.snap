// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoodDeed component renders Matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="goodDeed"
  >
    <div
      class="rounded-lg mt-2 shadow-lg bg-white	 w-full flex flex-row flex-wrap p-3 antialiased"
    >
      <div
        class="md:w-2/3 w-full px-3 flex flex-row flex-wrap"
      >
        <div
          class="w-full text-gray-700 font-semibold relative pt-3 md:pt-0"
        >
          <div
            class="flex flex-row text-2xl text-amber-500 leading-tight pb-1"
          >
             //! get good deed title Good Deed Name
          </div>
          <div
            class="text-sm text-amber-500 md:absolute pt-3 md:pt-0 top-0 right-0"
          >
            //!we need to be able to add this to the users total on verification Kindly Points: 
            <b>
              +5
            </b>
          </div>
          <div
            class="text-normal hover:text-cyan-700 cursor-pointer text-cyan-900 pb-4"
          >
            //! get good deed host
            <span
              class="pb-1"
            >
              User Host
            </span>
          </div>
          <div
            class="text-normal text-cyan-900 pb-1"
          >
            <span
              class=""
            >
              //! get good deed description
              <b>
                Description:
              </b>
               lorem fdjasfkldsjafkldsjafl fdjsalfkjdsalf dasfjdklsafjdklsaf 
            </span>
          </div>
          <div
            class="text-normal text-cyan-900 pb-1"
          >
            <span
              class=""
            >
              //! get good deed location
              <b>
                Location:
              </b>
              Tolland, CT
            </span>
          </div>
          <div
            class="text-normal text-cyan-900 pb-1"
          >
            <span
              class=""
            >
              //! get good deed dateTime
              <b>
                Date:
              </b>
               04/22/2122
            </span>
          </div>
          <div
            class="text-normal text-cyan-900 pb-1"
          >
            <span
              class=""
            >
              <b>
                Time:
              </b>
               1pm-3pm
            </span>
          </div>
          <div
            class="text-sm text-amber-500 md:absolute pt-3 md:pt-0 bottom-0 right-0"
          >
            <button
              class="bg-cyan-700  hover:bg-orange-300 text-white font-bold py-2 px-4 rounded mt-1"
            >
              Be Kind //! need to add helper after signup
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
